oci/auth/aws/auth.go

```diff
@@ -34,17 +35,22 @@ import (
 	"github.com/fluxcd/pkg/oci"
 )
 
-var registryPartRe = regexp.MustCompile(`([0-9+]*).dkr.ecr.([^/.]*)\.(amazonaws\.com[.cn]*)`)
+var registryPartRe = regexp.MustCompile(`(^[a-zA-Z0-9][a-zA-Z0-9-_]*)\.dkr\.ecr(-fips)?\.([a-zA-Z0-9][a-zA-Z0-9-_]*)\.amazonaws\.com(\.cn)?$`)
 
 // ParseRegistry returns the AWS account ID and region and `true` if
 // the image registry/repository is hosted in AWS's Elastic Container Registry,
 // otherwise empty strings and `false`.
 func ParseRegistry(registry string) (accountId, awsEcrRegion string, ok bool) {
-	registryParts := registryPartRe.FindAllStringSubmatch(registry, -1)
+	registry = strings.TrimPrefix(registry, "https://")
+	registryURL, err := url.Parse("https://" + registry)
```

Comment:
`ref, _ := name.ParseReference(registry)` will return `index.docker.io` for `012345678901.dkr.ecr.us-east-1.amazonaws.com` because it does not have a repository and identifier part. 

This is a valid registry and a valid input to parse registry so it should be handled by the function in my opinion

Reply:
got it, I removed this logic from the PR